src/System.Collections.NonGeneric/tests/QueueTests.cs

```diff
@@ -479,6 +479,22 @@ public static void GetEnumerator(int count)
         }
 
         [Fact]
+        public static void GetEnumerator_IClonable_Clone()
+        {
+            Queue queue = Helpers.CreateIntQueue(10);
+
+            IEnumerator enumerator = queue.GetEnumerator();
+            ICloneable cloneableEnumerator = (ICloneable)enumerator;
+
+            IEnumerator clonedEnumerator = (IEnumerator)cloneableEnumerator.Clone();
+            Assert.NotSame(enumerator, clonedEnumerator);
+
+            // Cloned and original enumerators should enumerate separately.
+            while (enumerator.MoveNext()) ;
+            Assert.True(clonedEnumerator.MoveNext());
```

Comment:
Cheers, I've responded to your Pr feedback (made more sense in a squashed commit as most code is different)